infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/StreamsSnapshotWriter.java

```diff
@@ -214,7 +224,11 @@ private void updateLog(TxnContext txnContext, List<SMREntry> smrEntries, UUID st
 
         for (SMREntry smrEntry : smrEntries) {
             if (session.getSubscriber().getModel().equals(ROUTING_QUEUES)) {
-                txnContext.logUpdate(streamId, smrEntry, Collections.singletonList(replicatedRoutingQueueTag));
+                CorfuRecord<Queue.RoutingTableEntryMsg, Message> record =
```

Comment:
updateLog() is used for writing to shadow streams and actual streams.  I think we should re-generate the key only once out of the 2 phases.

Reply:
sure, I'll put this in the 'apply' path